src/support/dtypes.h

```diff
@@ -63,14 +63,16 @@
 */
 
 #ifdef _OS_WINDOWS_
-#define STDCALL __stdcall
+#define STDCALL  __stdcall
+#define JL_NAKED __attribute__ ((naked))
 # ifdef LIBRARY_EXPORTS
 #  define JL_DLLEXPORT __declspec(dllexport)
 # else
 #  define JL_DLLEXPORT __declspec(dllimport)
 # endif
 #else
 #define STDCALL
+#define JL_NAKED     __attribute__ ((naked))
 #define JL_DLLEXPORT __attribute__ ((visibility("default")))
```

Comment:
```suggestion
#define JL_DLLEXPORT __attribute__ ((visibility("protected")))
```

Need to change this to make linux act like the other operating systems